Molio/Source/Domain/Entity/MusicDeck/MusicDeck.swift

```diff
@@ -0,0 +1,10 @@
+import Foundation
+import Combine
+
+protocol MusicDeck {
+    func musicPublisher(at: Int) -> AnyPublisher<RandomMusic?, Never>
```

Comment:
이게 잘 이해가되지 않는데요, 왜 개별 음악에 대해서 `AnyPublisher` 를 반환하는 구조가 되었을까요?

Reply:
맨토님 귀한 시간 내어 읽어봐주셔서 감사합니다.

개별 음악에 대한 
```suggestion
protocol MusicDeck {
    var currentMusicCardPublisher: AnyPublisher<RandomMusic?, Never>
    var nextMusicCardPublisher: AnyPublisher<RandomMusic?, Never>
    
     private func musicPublisher(at: Int) -> AnyPublisher<RandomMusic?, Never>
```

말씀을 듣고 @alstn38  민수님께서 의견을 내어주셔서
추상적으로 느껴지는 `musicPublisher`로부터 퍼블리셔를 제공받기보다는 의미가 더 명확한 프로퍼티로 받기로 했습니다.

`currentMusicCardPublisher`는 현재 노래의 변경에 반응해야하는 객체들이 받아서 자신의 프로퍼티와 구독합니다.
예를 들면 `AudioPlayer`, `CurrentMusicCardView`가 퍼블리셔를 제공받고 현재 노래의 변경에 반응합니다.

혹시 이런 식으로 Publisher를 반환하는 구조 자체가 일반적이지 않은지 궁금하기도 합니다 ㅠㅠ
이따 맨토링 시간에 여쭤보고 싶은 내용입니다.

감사합니다!!